bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -1361,25 +1401,20 @@ public void shutdown() {
         // since logChannel is buffered channel, do flush when shutting down
         LOG.info("Stopping EntryLogger");
         try {
+            //close corresponding fileChannels for read
+            fileChannelBackingCache.closeAllFileChannels();
```

Comment:
Invalidating all the caches should have the same effect.

Reply:
If we want to invalidate the threadLocal cache, then we has 2 ways to implement which was discussed in here(<html>https://stackoverflow.com/questions/2795447/is-there-no-way-to-iterate-over-or-copy-all-the-values-of-a-java-threadlocal</html>).  I think invalidating method is complicated than simply closing in fileChannelBackingCache.

Reply:
it doesn't add much complexity. You just need to create a concurrent list of some sort, and when initialValue is called, add the cache to the list before you return it. The threads in bookkeeper live for the lifetime of the process, so you don't need to worry about removing the caches from the list (otherwise a per thread cache would be pointless).